### PR TITLE
chore: rework vm ressource fee

### PR DESCRIPTION
**Pull request by tdelabro**
update dependencies, including blockifier where we previously replaced float by FixedU128.
We also come back to an hashmap for the vm_ressource_fee_cost, for better integration with blockifier's type

### CODE REVIEW

File: crates/primitives/fee/src/lib.rs [hunk 1]
```diff
@@ -33,15 +38,33 @@ pub const FEE_TRANSFER_N_STORAGE_CHANGES: u8 = 2; // Sender and sequencer balanc
...
-    "pedersen_builtin" => 0.32_f64,
-    "range_check_builtin" => 0.16_f64,
-    "ecdsa_builtin" => 20.48_f64,
-    "bitwise_builtin" => 0.64_f64,
-    "poseidon_builtin" => 0.32_f64,
-    "ec_op_builtin" => 10.24_f64,
-};
+pub static VM_RESOURCE_FEE_COSTS: [(&'static str, FixedU128); 7] = [
+    ("n_steps", FixedU128::from_inner(10_000_000_000_000_000)),
+    ("pedersen_builtin", FixedU128::from_inner(32_000_000_000_000_000)),
```
Comment (d-roak) [inline crates/primitives/fee/src/lib.rs:43]:
```suggestion
    ("pedersen_builtin", FixedU128::from_inner(320_000_000_000_000_000)),
```

File: crates/primitives/fee/src/lib.rs [hunk 2]
```diff
@@ -33,15 +38,33 @@ pub const FEE_TRANSFER_N_STORAGE_CHANGES: u8 = 2; // Sender and sequencer balanc
...
+    ("range_check_builtin", FixedU128::from_inner(16_000_000_000_000_000)),
```
Comment (d-roak) [inline crates/primitives/fee/src/lib.rs:44]:
```suggestion
    ("range_check_builtin", FixedU128::from_inner(160_000_000_000_000_000)),
```

File: crates/primitives/fee/src/lib.rs [hunk 3]
```diff
@@ -33,15 +38,33 @@ pub const FEE_TRANSFER_N_STORAGE_CHANGES: u8 = 2; // Sender and sequencer balanc
...
+#[cfg(test)]
+mod vm_resource_fee_costs {
+    use super::{FixedU128, HashMap, VM_RESOURCE_FEE_COSTS};
+
+    #[test]
+    fn check_values_as_floats() {
+        let hm = HashMap::from(VM_RESOURCE_FEE_COSTS);
+
+        assert_eq!(hm.get("n_steps"), Some(FixedU128::from_float(0.01)).as_ref());
+        assert_eq!(hm.get("pedersen_builtin"), Some(FixedU128::from_float(0.032)).as_ref());
```
Comment (d-roak) [inline crates/primitives/fee/src/lib.rs:60]:
```suggestion
        assert_eq!(hm.get("pedersen_builtin"), Some(FixedU128::from_float(0.32)).as_ref());
```

File: crates/primitives/fee/src/lib.rs [hunk 4]
```diff
@@ -33,15 +38,33 @@ pub const FEE_TRANSFER_N_STORAGE_CHANGES: u8 = 2; // Sender and sequencer balanc
...
+        assert_eq!(hm.get("range_check_builtin"), Some(FixedU128::from_float(0.016)).as_ref());
```
Comment (d-roak) [inline crates/primitives/fee/src/lib.rs:61]:
```suggestion
        assert_eq!(hm.get("range_check_builtin"), Some(FixedU128::from_float(0.16)).as_ref());
```

**Review by d-roak — COMMENTED** [review]
The suggestions are based on the previous values

File: crates/primitives/fee/src/lib.rs (hunk 1 above)
Comment (tdelabro) [reply crates/primitives/fee/src/lib.rs]: done

File: crates/primitives/fee/src/lib.rs (hunk 2 above)
Comment (tdelabro) [reply crates/primitives/fee/src/lib.rs]: done

File: crates/primitives/fee/src/lib.rs (hunk 3 above)
Comment (tdelabro) [reply crates/primitives/fee/src/lib.rs]: done

File: crates/primitives/fee/src/lib.rs (hunk 4 above)
Comment (tdelabro) [reply crates/primitives/fee/src/lib.rs]: done

**Review by d-roak — APPROVED** [review]
lgtm

**Pull request merged**